### PR TITLE
Add slot support

### PR DESCRIPTION
**Pull request by KieranChandler**

### CODE REVIEW

File: src/Farmer/Arm/Web.fs [hunk 1]
```diff
@@ -258,4 +271,42 @@ module SiteExtensions =
...
\ No newline at end of file
+                siteExtensions.Create(this.SiteName/this.Name, this.Location, [ sites.resourceId this.SiteName ]) :> _
+
+type Slot =
+    { SlotName: string
+      Location : Location
+      ServicePlan: ResourceId
+      Site: ResourceId
+      Tags: Map<string,string>
+      AutoSwapSlotName: string
```
Comment (TheRSP) [inline src/Farmer/Arm/Web.fs:282]: Should this be a `string Option` to allow disabling auto slot swap?

File: src/Farmer/Builders/Builders.Functions.fs [hunk 2]
```diff
@@ -159,6 +275,17 @@ type FunctionsConfig =
...
             | None ->
                 ()
+                
+            for kvp in this.Slots do
+                let name,cfg = kvp.Key,kvp.Value
+                { SlotName = name 
+                  Location = location
+                  ServicePlan = this.ServicePlanId
+                  Site = this.ResourceId
+                  Tags = this.Tags
```
Comment (TheRSP) [inline src/Farmer/Builders/Builders.Functions.fs:285]: Should we allow slots to have their own tags? and if so, do we merge with the web app slot tags?

File: src/Farmer/Builders/Builders.Functions.fs [hunk 3]
```diff
@@ -159,6 +275,17 @@ type FunctionsConfig =
...
+                  AutoSwapSlotName = kvp.Value.AutoSwapSlotName
```
Comment (TheRSP) [inline src/Farmer/Builders/Builders.Functions.fs:286]:
```suggestion
                  AutoSwapSlotName = cfg.AutoSwapSlotName
```

**Review by TheRSP — APPROVED** [review]

File: src/Farmer/Builders/Builders.Functions.fs (hunk 2 above)
Comment (KieranChandler) [reply src/Farmer/Builders/Builders.Functions.fs:285]: I don't think at this point in the code we should make a decision like that. Would that be a codat thing or do you think it just makes sense in general?

File: src/Farmer/Arm/Web.fs [hunk 4]
```diff
@@ -258,4 +268,42 @@ module SiteExtensions =
...
+                | ParameterSetting s -> Some s
+                | ExpressionSetting _ | LiteralSetting _ -> None)
+    interface IArmResource with
+        member this.ResourceId = ResourceId.create (slots, this.ResourceName)
+        member this.JsonModel =
+            {| slots.Create(this.ResourceName, this.Location, [this.Site], this.Tags) with
+                 properties = 
+                    {| serverFarmId = this.ServicePlan.ArmExpression.Eval()
+                       siteConfig = 
+                        {| autoSwapSlotName = this.AutoSwapSlotName
```
Comment (TheRSP) [inline src/Farmer/Arm/Web.fs:298]: Does this work nicely with serialisation? Does it serialise Option nicely?

**Pull request merged**